enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,494 @@
+
+In OpenShift 3.x the file-based policies were customizable by customers. In 4.x, with its unclear future and its complexity involved, 
+we hesitate to go the same route: if the file-based API goes away, it is infeasible to patch it back into our OpenShift API server
+binaries without major effort.
+
+From the requirements we get from customers, it is apparent that the file-based policies are far too low-level, and
+too complex to get right. This enhancement is about this high-level API which we can offer to satisfy the needs of
+customers with strong regulatory or security requirement, while being easy to configure for them and feasible to 
+maintain in the future.
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### Goals
+
+- add an abstract audit policy configuration to `apiservers.config.openshift.io/v1`
+  - with few predefined policies
+  - easily extensible in the future 
+  - few knobs on the level of concrete OpenShift customer requirement
+- applied by kube-apiserver, openshift-apiserver and oauth-apiserver.
+
+### Non-Goals
+
+- configuration of audit backends like webhooks
+- configuration of file logger parameters
+
+## Proposal
+
+We propose to add a `audit.profile` field to `apiservers.config.openshift.io/v1` which selects the 
+audit policy to be deployed to all OpenShift-provided APIservers in the cluster. From the beginning
+we provide the following profile:
+
+- `Default`: this is the existing default policy. `Default` is also the default value in the `APIServer` custom resource:
+
+  ```yaml
+    apiVersion: audit.k8s.io/v1beta1
+    kind: Policy
+    # Don't generate audit events for all requests in RequestReceived stage.
+    omitStages:
+    - "RequestReceived"
+    rules:
+    # Don't log requests for events
+    - level: None
+      resources:
+      - group: ""
+        resources: ["events"]
+    # Don't log oauth tokens as metadata.name is the secret
+    - level: None
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+    # Don't log authenticated requests to certain non-resource URL paths.
+    - level: None
+      userGroups: ["system:authenticated", "system:unauthenticated"]
+      nonResourceURLs:
+      - "/api*" # Wildcard matching.
+      - "/version"
+      - "/healthz"
+      - "/readyz"
+  
+    # Here the tail begins which differs per profile:
+  
+    # A catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - "RequestReceived"
+   ```
+
+- `OauthUserRequests`: this is like `Default`, but it logs request bodies for requests for oauth users only, i.e. by users outside of the cluster.
```

Comment:
this is technically useful. users would create service accounts and use those. This doesn't buy value.  if we want "log bodies" make it "log bodies", but don't try exclusion.  We should be heavily biased towards post-filtering